packages/auth/ios/RNFBAuth/RNFBAuthModule.m

```diff
@@ -939,6 +941,181 @@ - (void)invalidate {
+    {
+        FIRPhoneAuthCredential *credential = [
+                                         [FIRPhoneAuthProvider providerWithAuth:[FIRAuth authWithApp:firebaseApp]]
+                                         credentialWithVerificationID:verificationId
+                                         verificationCode:kPhoneSecondFactorVerificationCode
+                                         ];
+        
+        FIRMultiFactorAssertion *assertion = [FIRPhoneMultiFactorGenerator assertionWithCredential:credential];
+        // Complete enrollment. This will update the underlying tokens
+        // and trigger ID token change listener.
+        [user.multiFactor enrollWithAssertion:assertion
+                                  displayName:displayName
+                                   completion:^(NSError * _Nullable error)
+        {
+            if (error) {
+                [self promiseRejectAuthException:reject error:error];
+            } else {
+                [self promiseWithUser:resolve rejecter:reject user:user];
+            }
+        }];
+    } else {
+        [self promiseNoUser:resolve rejecter:reject isError:YES];
+    }
+}
+     
+RCT_EXPORT_METHOD(signInWithMultiFactorInfo:
+                  (FIRApp *) firebaseApp
+                  :(NSString *) multiFactorHintUID
+                  :(RCTPromiseResolveBlock) resolve
+                  :(RCTPromiseRejectBlock) reject
+                  )
+{
+    FIRMultiFactorResolver *resolver = [mfaResolvers objectForKey:multiFactorHintUID];
+    FIRMultiFactorInfo *hint = [resolver.hints objectAtIndex:[resolver.hints indexOfObjectPassingTest:^BOOL(FIRMultiFactorInfo * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+        return [obj.UID isEqual:multiFactorHintUID];
+    }]];
+    
+    if ([hint.factorID isEqual:@"1"]  || YES) { // Currently Firebase only supports phone MFA
+        // Send SMS verification code
+        [
+         [FIRPhoneAuthProvider providerWithAuth:[FIRAuth authWithApp:firebaseApp]] verifyPhoneNumberWithMultiFactorInfo:(FIRPhoneMultiFactorInfo*) hint
+          UIDelegate:nil
+          multiFactorSession:resolver.session
+          completion:^(NSString * _Nullable verificationID, NSError * _Nullable error) {
+            if (error != nil) {
+                [self promiseRejectAuthException:reject error:error];
+            } else {
+                [mfaResolvers setObject:resolver forKey:verificationID];
+                NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+                [defaults setObject:verificationID forKey:@"authVerificationID"];
+                resolve(@{
+                    @"verificationId": verificationID
+                        });
+            }
+          }
+        ];
+    } else {
+        [RNFBSharedUtils rejectPromiseWithUserInfo:reject userInfo:(NSMutableDictionary *) @{
+            @"code": @"unsupported-second-factor",
+            @"message": @"This second factor is not supported",
+        }];
+    }
```

Comment:
same fast-error suggestion here - do this error handling first up top and return, then rest of the code is clean mentally from the extra indent / mental load